### PR TITLE
refactor: simplify NFTView container

### PR DESCRIPTION
**Pull request by joonchoi0918**
[Associated Notion Card](https://www.notion.so/Refactor-NFTView-component-8c119c46cd274464b42b61d452da4596)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

### CODE REVIEW

File: src/containers/other/NFTView/hooks/useNftData.ts
Comment (ericlowturing) [inline src/containers/other/NFTView/hooks/useNftData.ts:86]: 👍

File: src/containers/other/NFTView/hooks/useNftData.ts
Comment (ericlowturing) [inline src/containers/other/NFTView/hooks/useNftData.ts:100]: Oh! 💯

File: src/containers/other/NFTView/hooks/useNftData.ts
Comment (ericlowturing) [inline src/containers/other/NFTView/hooks/useNftData.ts:149]: 👍

File: src/containers/other/NFTView/hooks/useNftData.ts
Comment (ericlowturing) [inline src/containers/other/NFTView/hooks/useNftData.ts:171]: 👍

File: src/containers/other/NFTView/NFTView.tsx
Comment (ericlowturing) [inline src/containers/other/NFTView/NFTView.tsx:18]: Oh this is BIG change of large component. Nice working @joonchoi0918

File: src/containers/other/NFTView/NFTView.types.ts
Comment (ericlowturing) [inline src/containers/other/NFTView/NFTView.types.ts:12]: Nice union types! Btw there is not this type defined in SDK? @cc @colbr

**Review by ericlowturing — COMMENTED** [review]
LGTM. Nice refactoring. I left a few comments @joonchoi0918

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (colbr) [reply src/containers/other/NFTView/NFTView.helpers.ts:57]: Agreed, I like this more than the string concatenation we were doing before

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (colbr) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx:116]: There's a library function called `truncateAddress` which should serve this purpose

File: src/containers/other/NFTView/NFTView.types.ts
Comment (colbr) [reply src/containers/other/NFTView/NFTView.types.ts:12]: I don't think so! At least not in the current version

**Review by colbr — CHANGES_REQUESTED** [review]
Looking really nice, good job. The only required changes are the `Histories` ones I mentioned in feedback :)

I'm also about to push a few changes to `NFTView.tsx` in `develop` which will probably give you some conflicts - I'm happy to resolve them

File: src/config.ts
Comment (joonchoi0918) [reply src/config.ts:3]: Yeah, it works as I expected.

File: src/containers/other/NFTView/NFTView.constants.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.constants.tsx:4]: No worries. It works 👍

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts:39]: Will fix it.

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts:57]: I prefer to use REGEX and it makes code cleaner

File: src/containers/other/NFTView/elements/Attributes/Attributes.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Attributes/Attributes.tsx:82]: I just copied the JSX from old NFTView.tsx, let me update scss and remove inline styles

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx:111]: Ok, constants! Will fix it

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx:116]: Yeah, I just moved this component folder. Will look and fix it. Thanks

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx:18]: Agreed! Switch Case!

File: src/containers/other/NFTView/elements/Stats/Stats.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Stats/Stats.tsx:65]: No, It is not dependency

File: src/containers/other/NFTView/elements/Stats/Stats.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Stats/Stats.tsx:72]: I don't agree. It does not work in this case

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts:30]: Yeah, I created it

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts:124]: Will look and fix it. Thanks

Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts]: @colbr I have updated

File: src/containers/other/NFTView/NFTView.helpers.ts
Comment (joonchoi0918) [reply src/containers/other/NFTView/NFTView.helpers.ts]: Done

File: src/containers/other/NFTView/elements/Attributes/Attributes.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Attributes/Attributes.tsx]: Done

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx]: Done

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx]: Done

File: src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx
Comment (joonchoi0918) [reply src/containers/other/NFTView/elements/Histories/elements/HistoryItem.tsx]: Done

**Review by colbr — APPROVED** [review]

**Comment by colbr** [comment]
Looking good, thanks for making the requested changes! Will hold off on merging this for a couple of days so we can get the major release out. I will handle any merge conflicts that pop up

**Comment by colbr** [comment]
Pointing this PR to another branch to merge with some other refactoring, then will merge to `develop` when ready 😊

**Pull request merged**
